GCMtools/utils/interface.py

```diff
@@ -10,27 +10,344 @@
+
+            ds = ds.interp(lon=lons, lat=lats)
+
+        self.ds = ds
+        self.chemistry.set_data(ds)
+
+    def chem_from_poorman(self, temp_key='T', CO=0.55, FeH=0.0):
+        """
+        Calculate equilibrium abundancies with poorman from pRT
+
+        Parameters
+        ----------
+        temp_key: str, optional
+            The key to the temperature field used for the abundancies.
+            Defaults to T.
+        CO: float, optional
+            The C/O ratio. Currently only one global value allowed. Defaults to 0.55.
+        FeH: float, optional
+            The metalicity ratio. Currently only one global value allowed. Defaults to 0.0.
+        """
+        return self.chemistry.chem_from_poorman(temp_key=temp_key, CO=CO, FeH=FeH)
+
+
+
+class PACInterface(Interface):
     def to_2D_PAC(self, ds):
         """
         Transform the given GCM dataset to a pseudo-2D PAC
         input file.
 
         Parameters
         ----------
-        ds : DataSet
+        ds: DataSet
             A GCMtools-compatible dataset of a 3D climate simulation.
+
         """
-        # TODO: implement later
         pass
+
+
+class pRTInterface(Interface):
+    """
+    Interface with petitRADTRANS
+    """
+
+    def __init__(self, pRT):
+        """
+        Constructs the Interface and links to pRT.
+
+        Parameters
+        ----------
+        pRT: petitRADTRANS.Radtrans
+            A pRT Radtrans object
+        """
+        self.pRT = pRT
+
+    def calc_phase_spectrum(self, mmw, Rstar, Tstar, semimajoraxis, gravity=None, filename=None, normalize=True, **prt_args):
+        """
+        Calculate the spectrum for a phasecurve.
+
+        Parameters
+        ----------
+        mmw: float or 1D-array
+            Mean molecular weight (in atomic units). Will be globally uniform if float or horizonatally uniform if 1D.
+        Rstar: float
+            stellar radius in !cm!.
+        Tstar: float
+            Temperature of the hoststar.
+        semimajoraxis: float
+            The distance between hoststar and planet in !cm!
+        gravity: float
+            surface gravity in !cgs!. Will default to the value provided by GCMT.
+        filename: str
+            path at which the output should be stored.
+        normalize: bool
+            Decide if you want the spectrum to be normalized to the star. The code will then:
+            1. correct intensity for the ratio between solar radius and planetary radius
+            2. devide by stellar spectrum
+        prt_args:
+            All the args that should be parsed to calc_spectra.
+            See the docs of prt_phasecurve for more info on the arguments.
+
+        Returns
+        -------
+        spectra: xr.DataArray
+            Dataarray containing the spectrum. The spectrum is normed to the stellar spectrum.
+
+        """
+        from prt_phasecurve import calc_spectra
+        import petitRADTRANS.nat_cst as nc
```

Comment:
Not sure if it is best practice to import them here or for the file in general. Personally, I don't mind much either way, but this is something we might want to look up.